src/message/messageReducers.js

```diff
@@ -0,0 +1,157 @@
+/* @flow */
+import omit from 'lodash.omit';
+
+import type {
+  MessagesState,
+  MessageAction,
+  MessageFetchCompleteAction,
+  EventReactionAddAction,
+  EventReactionRemoveAction,
+  EventNewMessageAction,
+  EventMessageDeleteAction,
+  EventUpdateMessageAction,
+} from '../types';
+import {
+  APP_REFRESH,
+  LOGOUT,
+  LOGIN_SUCCESS,
+  ACCOUNT_SWITCH,
+  MESSAGE_FETCH_COMPLETE,
+  EVENT_NEW_MESSAGE,
+  EVENT_MESSAGE_DELETE,
+  EVENT_REACTION_ADD,
+  EVENT_REACTION_REMOVE,
+  EVENT_UPDATE_MESSAGE,
+} from '../actionConstants';
+import { NULL_ARRAY, NULL_OBJECT } from '../nullObjects';
+import { groupItemsById } from '../utils/misc';
+
+const initialState: MessagesState = NULL_OBJECT;
+
+const messageFetchComplete = (
+  state: MessagesState,
+  action: MessageFetchCompleteAction,
+): MessagesState => ({
+  ...state,
+  ...groupItemsById(action.messages),
+});
+
+const eventReactionAdd = (state: MessagesState, action: EventReactionAddAction): MessagesState => {
+  const oldMessage = state[action.message_id];
+  if (!(action.message_id in state)) {
+    return state;
+  }
```

Comment:
Style nit: I'd define `oldMessage` after you've done the `action.message_id in state` check, rather than before.

Unless you use it to *do* that check. I think `oldMessage === undefined`, or even `!oldMessage`, would be fine for that.